### PR TITLE
HimmelKreis4865 kick

### PR DESCRIPTION
**Pull request by HimmelKreis4865**
- [x] This PR actually submits something - don't use this system as a messaging service!
- [x] This PR isn't duplicated - you can check if it is by scanning the small list - link is on the navigation bar for this repository.
- [x] This PR includes appropriate markdown for sections - e.g. code blocks for suggested code.
- [x] This PR description and comments in code is understandable - feel free to use your native language to write if you are not comfortable with English.
- [x] This PR has a single branch for itself in your fork - don't just commit to the master branch of your repository!
- [x] This PR has comments written in clear English - please don't write unreadable comments just for your eyes.

## Pull Request description
This PR adds KickSubCommand ( /p kick ) to kick players from your plot.

## Optional information
There is a translation for english and german, but not for the others.

### CODE REVIEW

File: resources/deu.ini [hunk 1]
```diff
@@ -227,4 +227,15 @@ clone.desc=Klont ein Grundstück an eine andere Position
...
+; the /p kick Subcommand
+kick.name=kick
+kick.desc=Kicke einen Spieler von deinem Plot
+kick.usage=/p kick <player: target>
+kick.noPlayer=Der angegebene Spieler konnte nicht gefunden werden
+kick.notInPlot=Der angegebene Spieler befindet sich nicht in deinem Plot
+kick.cannotkick=Du kannst diesen Spieler nicht kicken
+kick.attemptkick=Der Spieler {%0} hat versucht, dich von seinem plot zu kicken
+kick.success1=Du hast erfolgreich {%0} von dem Plot {%1} gekickt
+kick.success2=Der Spieler {%0} hat dich von dem Plot {%1} gekickt
```
Comment (jasonwynn10) [inline resources/deu.ini:241]: Line endings should be CR, not CRLF.

File: src/MyPlot/Commands.php [hunk 2]
```diff
@@ -236,4 +238,4 @@ public function execute(CommandSender $sender, string $alias, array $args) : boo
 		}
 		return true;
 	}
-}
\ No newline at end of file
+}
```
Comment (jasonwynn10) [inline src/MyPlot/Commands.php:241]: Line endings should be CR, not CRLF.

File: src/MyPlot/subcommand/KickSubCommand.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        if ($target->teleport($this->getPlugin()->getPlotPosition($plot)->add(0, 2, 0)->subtract(1, 0, 1))) {
+            $sender->sendMessage($this->translateString("kick.success1", [$target->getName(), $plot->__toString()]));
+            $target->sendMessage($this->translateString("kick.success2", [$sender->getName(), $plot->__toString()]));
+            return true;
+        }
+        $sender->sendMessage($this->translateString("error"));
+        return true;
+    }
+}
```
Comment (jasonwynn10) [inline src/MyPlot/subcommand/KickSubCommand.php:60]: Line endings should be CR, not CRLF.

File: src/MyPlot/Commands.php [hunk 4]
```diff
@@ -75,6 +76,7 @@ public function __construct(MyPlot $plugin) {
 		$this->loadSubCommand(new SetOwnerSubCommand($plugin, "setowner"));
 		$this->loadSubCommand(new ListSubCommand($plugin, "list"));
 		$this->loadSubCommand(new PvpSubCommand($plugin, "pvp"));
+        	$this->loadSubCommand(new KickSubCommand($plugin, "kick"));
```
Comment (jasonwynn10) [inline src/MyPlot/Commands.php:79]: Indentation needs fixed

File: src/MyPlot/subcommand/KickSubCommand.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
+<?php
+declare(strict_types=1);
+namespace MyPlot\subcommand;
+
+use pocketmine\command\CommandSender;
+use pocketmine\level\Position;
+use pocketmine\Player;
+use pocketmine\utils\TextFormat;
+
```
Comment (jasonwynn10) [inline src/MyPlot/subcommand/KickSubCommand.php:9]: File should use tabs for formatting, not spaces.

File: src/MyPlot/subcommand/KickSubCommand.php [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        if (!$this->getPlugin()->getPlotByPosition($target)->isSame($plot)) {
+            $sender->sendMessage(TextFormat::RED . $this->translateString("kick.notInPlot"));
+            return true;
+        }
+        if ($target->hasPermission("myplot.admin.kick.bypass")) {
+            $sender->sendMessage(TextFormat::RED . $this->translateString("kick.cannotkick"));
+            $target->sendMessage($this->translateString("kick.attemptkick", [$target->getName()]));
+            return true;
+        }
+        if ($target->teleport($this->getPlugin()->getPlotPosition($plot)->add(0, 2, 0)->subtract(1, 0, 1))) {
```
Comment (jasonwynn10) [inline src/MyPlot/subcommand/KickSubCommand.php:52]: use the `teleportPlayerToPlot()` function instead.

**Comment by HimmelKreis4865** [comment]
I created them one more time in UNIX line seperator, I can't do anything else now. 😞

**Comment by jasonwynn10** [comment]
Only two fixes left for `KickSubCommand.php`

**Comment by HimmelKreis4865** [comment]
Now, I changed to LF, replaced spaces with tabs and used `MyPlot::getInstance()->teleportPlayerToPlot(Player, Plot)`

**Pull request merged**